wave_vision/src/matcher/brute_force_matcher.cpp

```diff
@@ -53,28 +65,133 @@ void BruteForceMatcher::checkConfiguration(
         throw std::invalid_argument(
           "Norm type is not one of the acceptable values!");
     }
+
+    // Check the value of the ratio_test heuristic
+    if (check_config.ratio_threshold < 0.0 ||
+        check_config.ratio_threshold > 1.0) {
+        throw std::invalid_argument(
+          "ratio_threshold is not an appropriate value!");
+    }
+
+    // Check the value of the threshold distance heuristic
+    if (check_config.distance_threshold < 0) {
+        throw std::invalid_argument("distance_threshold is a negative value!");
+    }
+
+    // Only acceptable values are 1, 2, 4, and 8
+    if (check_config.fm_method < cv::FM_7POINT ||
+        check_config.fm_method > cv::FM_RANSAC) {
+        throw std::invalid_argument("fm_method is not an acceptable value!");
+    }
+
+    if (check_config.fm_method == 3 ||
+        (check_config.fm_method > cv::FM_LMEDS &&
+         check_config.fm_method < cv::FM_RANSAC)) {
+        throw std::invalid_argument("fm_method is not an acceptable value!");
+    }
+}
+
+std::vector<cv::DMatch> BruteForceMatcher::filterMatches(
+  std::vector<cv::DMatch> &matches) const {
+    std::vector<cv::DMatch> filt_matches;
+    float min_distance;
+
+    // Sort matches by distance in increasing order
+    std::sort(matches.begin(), matches.end());
+
+    min_distance = matches.begin()->distance;
+
+    // Keep any match that is less than the rejection heuristic times minimum
+    // distance
+    for (auto &match : matches) {
+        if (match.distance <=
+            this->current_config.distance_threshold * min_distance) {
+            filt_matches.push_back(match);
+        }
+    }
+
+    return filt_matches;
+}
+
+std::vector<cv::DMatch> BruteForceMatcher::filterMatches(
+  std::vector<std::vector<cv::DMatch>> &matches) const {
+    std::vector<cv::DMatch> filt_matches;
+    float ratio;
+
+    for (auto &match : matches) {
+        // Calculate ratio between two best matches. Accept if less than
+        // ratio heuristic
+        ratio = match[0].distance / match[1].distance;
+        if (ratio <= this->current_config.ratio_threshold) {
+            filt_matches.push_back(match[0]);
+        }
+    }
+
+    return filt_matches;
+}
+
+std::vector<cv::DMatch> BruteForceMatcher::removeOutliers(
+  const std::vector<cv::DMatch> &matches,
+  const std::vector<cv::KeyPoint> &keypoints_1,
+  const std::vector<cv::KeyPoint> &keypoints_2) const {
+    std::vector<cv::DMatch> good_matches;
+    std::vector<cv::Point2f> fp1, fp2;
+
+    // Take all good keypoints from matches, convert to cv::Point2f
+    for (auto &match : matches) {
+        fp1.push_back(keypoints_1.at(match.queryIdx).pt);
+        fp2.push_back(keypoints_2.at(match.trainIdx).pt);
+    }
+
+    // Find fundamental matrix
+    std::vector<uchar> mask;
+    cv::Mat fundamental_matrix;
+
+    fundamental_matrix =
+      cv::findFundamentalMat(fp1, fp2, cv::FM_RANSAC, 3., 0.99, mask);
```

Comment:
Fixed. They're parameters to specify the outlier threshold and confidence of the RANSAC estimation. As discussed below, they've been moved into the struct.